psi4/src/psi4/libfock/snLinK.cc

```diff
@@ -0,0 +1,498 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "jk.h"
+#include "SplitJK.h"
+#include "psi4/libqt/qt.h"
+#include "psi4/libfock/cubature.h"
+#include "psi4/libfock/points.h"
+#include "psi4/libmints/vector.h"
+#include "psi4/libmints/basisset.h"
+#include "psi4/libmints/electrostatic.h"
+#include "psi4/libmints/mintshelper.h"
+#include "psi4/libmints/molecule.h"
+#include "psi4/libmints/integral.h"
+#include "psi4/libmints/petitelist.h"
+#include "psi4/liboptions/liboptions.h"
+#include "psi4/lib3index/dftensor.h"
+#include "psi4/libpsi4util/PsiOutStream.h"
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <tuple>
+#include <unordered_set>
+#include <variant>
+#include <vector>
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+using namespace psi;
+
+namespace psi {
+
+// creates maps for translating Psi4 option inputs to GauXC enums
+std::tuple<
+    std::unordered_map<std::string, GauXC::PruningScheme>, 
+    std::unordered_map<std::string, GauXC::RadialQuad> 
+> snLinK::generate_enum_mappings() {
+    // generate map for grid pruning schemes 
+    std::unordered_map<std::string, GauXC::PruningScheme> pruning_scheme_map; 
+    pruning_scheme_map["ROBUST"] = GauXC::PruningScheme::Robust;
+    pruning_scheme_map["TREUTLER"] = GauXC::PruningScheme::Treutler;
+    pruning_scheme_map["NONE"] = GauXC::PruningScheme::Unpruned;
+
+    // generate map for radial quadrature schemes 
+    std::unordered_map<std::string, GauXC::RadialQuad> radial_scheme_map; 
+    radial_scheme_map["TREUTLER"] = GauXC::RadialQuad::TreutlerAldrichs;
+    radial_scheme_map["MURA"] = GauXC::RadialQuad::MuraKnowles;
+    // The Murray, Handy, Laming literature reference is mentioned in cubature.cc
+    // with association to this keyword
+    radial_scheme_map["EM"] = GauXC::RadialQuad::MurrayHandyLaming; 
+
+    // we are done
+    return std::make_tuple(pruning_scheme_map, radial_scheme_map);
+}
+
+// constructs a permutation matrix for converting matrices to and from GauXC's integral ordering standard 
+Eigen::PermutationMatrix<Eigen::Dynamic, Eigen::Dynamic> snLinK::generate_permutation_matrix(const std::shared_ptr<BasisSet> psi4_basisset) {
+  
+    Eigen::PermutationMatrix<Eigen::Dynamic, Eigen::Dynamic> permutation_matrix(psi4_basisset->nbf());
+    
+    // general array for how to reorder integrals 
+    constexpr int max_am = 10; 
```

Comment:
This should likely be derived from GauXC's max AM, not Libint's (although, ensuring that the two are compatible is not something that's currently being checked AFAIK). That is something you can get from GauXC (at least from the build system, I could also add it to `gauxc_config.hpp`).

Reply:
Aha, right, good catch! I remember when I wrote this, thinking I would change it later... and now it's later.

This should be addressed now! I also added a check to ensure that Libint2 and GauXC have the same internal AM limit as per @wavefunction91.

Reply:
Is the GauXC interface hardcoded to some value, or is the API independent of the size? In the latter case, the value should really be queried from the GauXC instance instead of compiling in a hardcoded value.

Reply:
As of yesterday, I use a very high hardcoded value for the AM here (11). But that has been changed in light of @TiborGY's comment; the API now queries the GauXC instance for the AM to use.